app/services/push_notifications/send.rb

```diff
@@ -0,0 +1,22 @@
+module PushNotifications
+  class Send
+    def self.call(user = nil, **kwargs)
+      new(user, **kwargs).call
+    end
+
+    def initialize(user, **kwargs)
+      @user = user
+      @title = kwargs[:title]
+      @body = kwargs[:body]
+      @payload = kwargs[:payload]
+    end
+
+    def call
+      @user.devices.each do |device|
+        device.create_notification(@title, @body, @payload)
+      end
+
+      PushNotifications::DeliverWorker.perform_in(30.seconds)
```

Comment:
`perform_in(30.seconds)` is key:
- The background worker will execute `Rpush.push` which will send out all pending notifications. This means we don't need to run it for **every PN**
- `PushNotifications::DeliverWorker` has a constraint that will execute the job only once (30 seconds from now) 

So if we need to send 1 PN every 5 seconds we only execute this once every 30s (no duplicate/unnecessary processing). If no PNs are scheduled to be sent for a 6h span then 0 jobs are executed.